addons/mail/models/mail_mail.py

```diff
@@ -348,11 +348,15 @@ def _send(self, auto_commit=False, raise_exception=False, smtp_session=None):
                     # see rev. 56596e5240ef920df14d99087451ce6f06ac6d36
                     notifs.flush(fnames=['notification_status', 'failure_type', 'failure_reason'], records=notifs)
 
+                # protect against ill-formatted email_from when formataddr was used on an already formatted email
+                emails_from = tools.email_split_and_format(mail.email_from)
+                email_from = emails_from[0] if emails_from else mail.email_from
```

Comment:
Could be inlined?

Reply:
It is used in a loop herebelow... we gain the if-else writing.